linker/jvm/src/main/scala/org/scalajs/linker/backend/closure/ClosureLinkerBackend.scala

```diff
@@ -96,17 +98,28 @@ final class ClosureLinkerBackend(config: LinkerBackendImpl.Config)
           closureExterns.asJava, List(module).asJava, options)
     }
 
+    if (result.errors.nonEmpty) {
+      throw new LinkingException(
+          "There were errors when applying the Google Closure Compiler")
+    }
+
     logger.time("Closure: Write result") {
       writeResult(topLevelVarDeclarations, result, compiler, output)
     }
   }
 
+  /** Constructs an externs file listing all the global refs.
+   */
+  private def makeExternsForGlobalRefs(globalRefs: Set[String]): String =
+    globalRefs.map("var " + _ + ";\n").mkString
+
   /** Constructs an externs file listing all exported properties in a linking
    *  unit.
    *
    *  This is necessary to avoid name clashes with renamed properties (#2491).
    */
-  private def makeExternsForExports(linkingUnit: LinkingUnit): String = {
+  private def makeExternsForExports(topLevelVarDeclarations: Option[String],
```

Comment:
I'm not sure I understand this: Why are *all* top level var declarations externs for exports? Aren't some top level var decls purely internal to the file?

Reply:
Because top-level var declarations are added after the fact *outside* of the IIFE that wraps the entire result of GCC. So from the point of view of GCC, all top-level exports are externs.

Recall that `topLevelVarDeclarations` is the set of names that are top-level-exported when in `NoModule`---empty otherwise. It was introduced in #3490.

Reply:
Right; so this actually makes perfect sense.